feature(api): add a defined over pressure error to prepare to aspirate

TamarZanzouri opened the pull request:
# Overview

partially closes [EXEC-557](https://opentrons.atlassian.net/browse/EXEC-557).
raise a defined over pressure error for prepare for aspirate command. 

## Test Plan and Hands on Testing
upload the following protocol, while pausing place the pad beneath the tip, resume and trigger an over pressure error.  

```
from opentrons import protocol_api, types


requirements = {
    "apiLevel": "2.18",
    "robotType": "Flex",
}


def run(protocol: protocol_api.ProtocolContext) -> None:
    magnetic_block = protocol.load_module(module_name="magneticBlockV1", location="D1")
    tip_rack = protocol.load_labware("opentrons_flex_96_tiprack_1000ul", "C1")
    well_plate = protocol.load_labware("nest_96_wellplate_100ul_pcr_full_skirt", "C3")

    protocol.load_trash_bin("A3")

    pipette = protocol.load_instrument("flex_1channel_1000", mount="left")

    pipette.pick_up_tip(tip_rack["A12"])
    pipette.move_to(well_plate["A1"].top())
    protocol.pause()
    pipette.prepare_to_aspirate()
```

## Changelog

add a defined over pressure error for `prepare_to_aspirate`


## Risk assessment

low.


[EXEC-557]: https://opentrons.atlassian.net/browse/EXEC-557?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

shlokamin committed:
fix(labware-library): make labware creator accessible via external links (#16480)

Fixes https://opentrons.atlassian.net/browse/RESC-339

TamarZanzouri committed:
fix(api): ER support for in place commands and blow out (#16510)

jerader committed:
feat(protocol-designer): mix tools advanced settings and mix batch edit (#16509)

closes AUTH-930 AUTH-926

mjhuff committed:
refactor(app): Update ignore error route in Error Recovery (#16511)

Partially closes EXEC-776

As the "ignore error and skip/retry" routing no longer applies only to one recovery option, it's worth revisiting the way we structure routing/copy. More importantly, we probably should avoid sending any sort of "ignore all errors" policy updates to the server until the user exits error recovery, because sending policy changes earlier could provide unexpected behavior if the user selects the "ignore all option" but then clicks "go back".

If the PUT request for the policy updates fail, route the user to our "recovery failed" modal and let the user select a different (or same) option.

TamarZanzouri committed:
prepare to aspirate over pressue

TamarZanzouri committed:
fix(api): add details to confusing error message (#16517)

sanni-t committed:
feat(api): add liquid class in PAPI (#16506)

AUTH-835, AUTH-836

# Overview

Adds `LiquidClass` class and `ProtocolContext.define_liquid_class()` to
PAPI. Also adds an `allowLiquidClasses` feature flag that will need to
be set to True in order to use the new API method.

## Changelog

- added `LiquidClass` dataclass to PAPI
- added `ProtocolContext.define_liquid_class` to define a liquid class
in the protocol
- added `allowLiquidClasses` internal feature flag

Some small schema changes:
- changed `liquidName` -> `liquidClassName` and made it a `safestring`
- added a `displayName`
- changed `tipType` -> tiprack

Misc shared data changes:
- added `liquid_classes.load_definition()`
- added validation tests for liquid class definitions

mjhuff committed:
feat(app): Add Error Recovery support for in-place commands (#16515)

Closes EXEC-777

#16510 appears to have fixed most of the inPlace issues we had seen back in 8.0, and it has (hopefully) greatly simplified the additional support the app needs to provide to get inPlace variations of aspirate/dispense working.

This commit just wires up those commands to the appropriate flows.

brenthagen committed:
feat(app): implement system language modal (#16507)

implements the system language modal that displays on first boot of the
desktop app or when the language of the computer running the desktop app
changes.

closes PLAT-536

koji committed:
fix(protocol-designer): add Eppendorf to removeOpentronsPhrases (#16470)

* fix(protocol-designer): add Eppendorf to removeOpentronsPhrases

koji committed:
fix(labware-library): fix labware-library css issue (#16502)

* fix(labware-library): fix labware-library css issue

y3rsh committed:
test(abt): fix Makefile variable (#16525)

pmoegenburg committed:
feat(api): add WellVolumeOffset to WellLocation (#16302)

<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure
you've read the "Opening Pull Requests" section of our Contributing
Guide:


https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams,
pictures, and videos along with text formatting make it possible to
create a rich and informative PR. For more information on GitHub
markdown, see:


https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out
the sections below to the best of your ability!
-->

# Overview

<!--
Describe your PR at a high level. State acceptance criteria and how this
PR fits into other work. Link issues, PRs, and other relevant resources.
-->

This PR enables static meniscus-relative aspirate and dispense via the
Protocol API. To enable this, WellVolumeOffset has been added to
WellLocation. This is a volume of liquid to account for when executing
commands with an origin of WellOrigin.MENISCUS. Specifying
`operationVolume` results in this class acting as a sentinel and should
be used when volume can be determined from the command parameters, for
example commanding Aspirate. A volume should be specified when it cannot
be determined from the command parameters, for example commanding
MoveToWell prior to AspirateInPlace.

## Test Plan and Hands on Testing

<!--
Describe your testing of the PR. Emphasize testing not reflected in the
code. Attach protocols, logs, screenshots and any other assets that
support your testing.
-->

Successfully tested the following protocol on a robot multiple times in
a row. Aspiration from meniscus was not tested due to not-yet-available
InnerWellGeometry:
```
from opentrons.protocol_api import ProtocolContext
metadata = {"protocolName": "Test LLD"}
requirements = {"robotType": "Flex", "apiLevel":"2.21"}


def run(ctx: ProtocolContext) -> None:
    """Run."""
    tiprack = ctx.load_labware(f"opentrons_flex_96_tiprack_1000uL", "A3")
    source = ctx.load_labware("nest_12_reservoir_15ml", "C2")
    sink = ctx.load_labware("nest_96_wellplate_100ul_pcr_full_skirt", "D2")
    pipette = ctx.load_instrument("flex_1channel_1000", "left", liquid_presence_detection = True)

    pipette.pick_up_tip(tiprack)
    pipette.measure_liquid_height(sink["A1"])
    pipette.aspirate(10, source["A1"])
    pipette.dispense(10, sink["A1"].meniscus(-2))
    pipette.return_tip()
```

## Changelog

<!--
List changes introduced by this PR considering future developers and the
end user. Give careful thought and clear documentation to breaking
changes.
-->

## Review requests

<!--
- What do you need from reviewers to feel confident this PR is ready to
merge?
- Ask questions.
-->

## Risk assessment

<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your
change may affect some other part of the system.
- For instance, changing return tip behavior may also change the
behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's
risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is
enough.
-->

sfoster1 reviewed (approved):
nice, thanks!

sfoster1 committed:
refactor(app-shell,app-shell-odd): update winston and improve logging (#16516)

Update [winston](https://www.npmjs.com/package/winston/v/3.15.0) which
is what we use for logging on the node side to the most recent version
(3.15) so that we can use `logger.child()`, which lets you override
logger metadata without having to create another logger instance.

The reason to do this is that each full winston logger instance hangs
event listeners off its shared transports
(https://github.com/winstonjs/winston/issues/1334) which results in
annoying node warning messages about memory leaks that aren't real and
are just based on "did you add more than a magic number of listeners to
this event". With the child logs, nothing adds the events, and we don't
get the warnings.

Also, get rid of the file logging to `/app/ODD/logs/error.log` and
`/app/ODD/logs/combined.log`, because we're already sending everything
to journald via the console and [using that to provide the logs via
http](https://github.com/Opentrons/opentrons/blob/edge/robot-server/robot_server/service/legacy/routers/logs.py#L16)
so it's just extra storage wear and space usage that we don't actually
need.

## testing
- [x] do the logs still go

SyntaxColoring commented on api/src/opentrons/protocol_engine/commands/prepare_to_aspirate.py:
```diff
@@ -29,25 +39,60 @@ class PrepareToAspirateResult(BaseModel):
-
-        return SuccessData(public=PrepareToAspirateResult(), private=None)
+        try:
+            current_position = await self._gantry_mover.get_position(params.pipetteId)
```
Style nitpick: I would keep this `get_position()` call outside the `try` block to make it very clear that the `try` block is only supposed to catch stuff from `prepare_for_aspirate()`.

It doesn't really matter in this case since it's only 2 lines anyway.

SyntaxColoring reviewed (approved):
Looks great, thanks!

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal edge snapshots (#16527)

The edge overnight analyses snapshot test is failing. This PR was opened
to alert us to the failure.

Co-authored-by: y3rsh <502770+y3rsh@users.noreply.github.com>

y3rsh committed:
fix(analyses-snapshot-testing): heal edge snapshots (#16526)

## Overview

Heal edge on missed snapshot change due to #16525

SyntaxColoring committed:
fix(api): Fix path planning after failed tip drop (#16513)

mjhuff committed:
refactor(app): update various error recovery copy/css (#16519)

Closes EXEC-776

This commit is a hodgepodge of small refactors to update error recovery to match designs (and fix small things that bother me and probably no one else).

b98064d - Updates the drop tip wizard copy to match designs
e8ff83f - If you look at "go back" and "continue" buttons, they aren't actually aligned. I think I borked this recently.
041305e - The deck map in ER slightly pushes down the "continue" button so it's not perfectly aligned with other error recovery "continue" buttons. This fixes that.
beb4184 - If you click "view error details" on the desktop app and adjust the size of your app, eventually the error details modal doesn't align with error recovery. Let's fix this by giving InterventionModal the same padding styling as the basic modal, which it should probably have regardless.

koji committed:
fix(protocol-designer): enable color picker's alpha value (#16505)

* fix(protocol-designer): enable color picker's alpha value

koji committed:
fix(protocol-designer): fix remove settings icon button (#16524)

* fix(protocol-designer): fix remove settings icon button

TamarZanzouri committed:
current position outside the try

koji committed:
fix(protocol-designer): fix create button clickable area issue (#16522)

* fix(protocol-designer): fix create button clickable area issue

SyntaxColoring committed:
refactor(api): Allow adding/removing tips on HW API without `await` (#16530)

TamarZanzouri committed:
change path to relative path

TamarZanzouri committed:
change path to relative path

TamarZanzouri committed:
Merge branch 'edge' into EXEC-557-overpressure-detected-during-prepare-to-aspirate

TamarZanzouri committed:
revert FE changes

TamarZanzouri committed:
revert test.json changes

mjhuff commented:
Merging this to unblock some FE work
